File: javascript/backup_dbs.js
#!/usr/bin/env node
// backup_dbs.py — portiert nach javascript
// Quelle: python, OpenClaw@gateway2:tmp/backup_dbs.py
// Erzeugt: 2026-08-08 durch ABSTRACTIONS_MANAGER.py

/** Backup docs.db and tree.db with timestamp into /workspace/db/backups */
const fs = require('fs');
const path = require('path');

const workspace = process.env.OPENCLAW_WORKSPACE || '/workspace';
const backupDir = path.join(workspace, 'db', 'backups');

// Ensure backupDir exists
fs.mkdirSync(backupDir, { recursive: true });

const timestamp = new Date().toISOString().slice(0, 19).replace(/:/g, '-').replace('T', '_');

['docs.db', 'tree.db'].forEach(dbName => {
    const src = path.join(workspace, dbName);
    if (fs.existsSync(src) && fs.statSync(src).isFile()) {
        const dest = path.join(backupDir, `${timestamp}_${dbName}.bak`);
        fs.copyFileSync(src, dest);
        console.log(`Backup created: ${dest}`);
    } else {
        console.log(`Source db not found: ${src}`);
    }
});

File: perl5/backup_dbs.pl
#!/usr/bin/perl
# backup_dbs.py — portiert nach perl5
# Quelle: python, OpenClaw@gateway2:tmp/backup_dbs.py
# Erzeugt: 2026-08-08 durch ABSTRACTIONS_MANAGER.py

use strict;
use warnings;
use File::Copy qw(copy);
use File::Path qw(make_path);
use POSIX qw(strftime);

# Backup docs.db and tree.db with timestamp into /workspace/db/backups

my $workspace = $ENV{'OPENCLAW_WORKSPACE'} // '/workspace';
my $backup_dir = "$workspace/db/backups";

# Ensure backup_dir exists
make_path($backup_dir) unless -d $backup_dir;

my $timestamp = strftime('%Y-%m-%d_%H-%M', localtime);

foreach my $db_name ('docs.db', 'tree.db') {
    my $src = "$workspace/$db_name";
    if (-f $src) {
        my $dest = "$backup_dir/${timestamp}_${db_name}.bak";
        copy($src, $dest) or warn "Copy failed: $!";
        print "Backup created: $dest\n";
    } else {
        print "Source db not found: $src\n";
    }
}
